dags/process_elife_zip_dag.py

```diff
@@ -0,0 +1,118 @@
+"""
+DAG processes a zip file stored in an AWS S3 bucket and prepares the extracted xml
+file for Libero content API and sends to the content store via a PUT request.
+"""
+import logging
+from datetime import timedelta
+from io import BytesIO
+from pathlib import Path
+from tempfile import TemporaryFile
+from zipfile import ZipFile
+
+from airflow import DAG, configuration
+from airflow.operators import python_operator
+from airflow.utils import timezone
+from lxml import etree
+
+from aws import get_aws_connection, list_bucket_keys_iter
+from task_helpers import get_previous_task_name
+
+ALLOWED_EXTENSIONS = {'.jpg', '.jpeg', '.pdf', '.xml'}
+
+SOURCE_BUCKET = configuration.conf.get('libero', 'source_bucket')
+DESTINATION_BUCKET = configuration.conf.get('libero', 'destination_bucket')
+
+logger = logging.getLogger(__name__)
+
+default_args = {
+    'owner': 'libero',
+    'depends_on_past': False,
+    'start_date': timezone.utcnow(),
+    'email': ['libero@example.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 3,
+    'retry_delay': timedelta(seconds=5)
+}
+
+
+def extract_zipped_files_to_bucket(**context):
```

Comment:
This module is specific for the .zip format, but I wonder if there is a more precise name we can use (article archive? article delivery?) to avoid the zip details leaking over code that is not tied to that. At the moment however, "zip" is practically not mentioned outside of this function, so no problem.

Reply:
renamed to `extract_archived_files_to_bucket`